### PR TITLE
lz4 1.9.2

### PR DESCRIPTION
**Pull request by jcfj**
Sorry, I forgot to update the version in https://github.com/mesonbuild/wrapweb/issues/103

I do have wraps for 1.9.1 (and 1.8.3) if you want, but I don't think that's meaningful.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+project('lz4', ['c'],
+  license: ['BSD', 'GPLv2'],
+  default_options : ['c_std=c99',
+    'buildtype=release'],
+  version: '1.9.2',
+  meson_version: '>=0.47.0')
+
+subdir('contrib/meson/meson')
+
+lz4_dep = liblz4_dep # Not sure this is a good idea…
```
Comment (sarum9in) [inline meson.build]: We usually have a style requirement of having `${wrap}_dep` exported, so I think it is.

File: contrib/meson/meson/examples/meson.build [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# All rights reserved.
+#
+# This source code is licensed under both the BSD-style license (found in the
+# LICENSE file in the root directory of this source tree) and the GPLv2 (found
+# in the COPYING file in the root directory of this source tree).
+# #############################################################################
+
+lz4_root_dir = '../../../..'
+
+#examples_c_args = ['-Wextra', '-Wundef', '-Wshadow', '-Wcast-align', '-Wstrict-prototypes']
```
Comment (sarum9in) [inline contrib/meson/meson/examples/meson.build:14]: Was it forgotten?

File: meson.build (hunk 1 above)
Comment (jcaesar) [reply meson.build]: Ty. *edited comment*

File: contrib/meson/meson/examples/meson.build (hunk 2 above)
Comment (jcaesar) [reply contrib/meson/meson/examples/meson.build:14]: Sorry, no idea. I kept those files unmodified from upstream and will delete when they're part of the next release.
https://github.com/lz4/lz4/blob/dev/contrib/meson/meson/examples/meson.build#L12

**Comment by jcaesar** [comment]
Mh, thinking about it, since I'm copying files from upstream, I guess I should include the LICENSE file, too?

**Comment by legeana** [comment]
It seems the upstream has different licenses. If necessary you should update `LICENSE.build` file accordingly.

From what I see [here](https://github.com/lz4/lz4/blob/dev/LICENSE#L2-L3)
```
- all files in the `lib` directory use a BSD 2-Clause license
- all other files use a GPLv2 license, unless explicitly stated otherwise
```

So it really depends where these files are coming from. As far as I can tell all files you included are dual licensed under BSD and GPLv2, and since BSD and MIT are compatible you can redistribute them under MIT as is.

**Comment by jcaesar** [comment]
> **If necessary** you should update LICENSE.build file accordingly.

If only I knew whether it is necessary… The way I read it is that anything that's not in `lib/` is GPLv2, hence so are the build files, hence BSD/MIT compatibility gets me nothing. (Does a meson_options.txt even reach the threshold of originality? I suppose it does?)

> So it really depends where these files are coming from.
* [LICENSE](https://github.com/lz4/lz4/blob/dev/lib/LICENSE)
* [COPYING](https://github.com/lz4/lz4/blob/dev/programs/COPYING)

**Comment by legeana** [comment]
`LICENSE.build` covers at least `upstream.build` and potentially any other files you add not covered by explicit licenses in headers. I would appreciate if you could keep it as is.

**Comment by jcaesar** [comment]
I have restored the `LICENSE.build`, but I'm confused. What is `upstream.build`? It's not a file that exists in this repository.
In fact, all files in this repository are GPLv2, which makes me think I should state so properly in the headers and get rid of any other license file?

**Comment by legeana** [comment]
`upstream.build` is the file you create for `wrapdb.mesonbuild.com`. It is not part of the original code you borrowed but rather a configuration you created for wrapdb consumption.

**Comment by jcfj** [comment]
?? 
Are you talking about [`/meson.build`](https://github.com/mesonbuild/lz4/blob/1.9.2/meson.build)? Because most of it (well, the `project` invocation) is also copied from upstream, and hence under GPLv2 as well.

**Comment by legeana** [comment]
No, I meant `upstream.wrap`, apologies for getting the suffix wrong.

**Comment by jcfj** [comment]
Ooooh! Thank you.

**Pull request merged**